src/Symfony/Bundle/FrameworkBundle/Routing/RouteCollectionBuilder.php

```diff
@@ -0,0 +1,393 @@
+    }
+
+    /**
+     * Sets a condition on all embedded routes (unless already set).
+     *
+     * @param string $condition
+     * @return $this
+     */
+    public function setCondition($condition)
+    {
+        $this->condition = $condition;
+
+        return $this;
+    }
+
+    /**
+     * Sets a default value that will be added to all embedded routes (unless that
+     * default value is already set.
+     *
+     * @param string $key
+     * @param mixed $value
+     * @return $this
+     */
+    public function setDefault($key, $value)
+    {
+        $this->defaults[$key] = $value;
+
+        return $this;
+    }
+
+    /**
+     * Sets a requirement that will be added to all embedded routes (unless that
+     * requirement is already set.
+     *
+     * @param string $key
+     * @param mixed $value
+     * @return $this
+     */
+    public function setRequirement($key, $regex)
+    {
+        $this->requirements[$key] = $regex;
+
+        return $this;
+    }
+
+    /**
+     * Sets an opiton that will be added to all embedded routes (unless that
+     * option is already set.
+     *
+     * @param string $key
+     * @param mixed $value
+     * @return $this
+     */
+    public function setOption($key, $value)
+    {
+        $this->options[$key] = $value;
+
+        return $this;
+    }
+
+    /**
+     * Sets the schemes on all embedded routes (unless already set).
+     *
+     * @param array|string $schemes
+     * @return $this
+     */
+    public function setSchemes($schemes)
+    {
+        $this->schemes = $schemes;
+
+        return $this;
+    }
+
+    /**
+     * Sets the methods on all embedded routes (unless already set).
+     *
+     * @param array|string $methods
+     * @return $this
+     */
+    public function setMethods($methods)
+    {
+        $this->methods = $methods;
+
+        return $this;
+    }
+
+    /**
+     * Adds a resource for this collection.
+     *
+     * @return $this
+     */
+    public function addResource(ResourceInterface $resource)
+    {
+        $this->resources[] = $resource;
+
+        return $this;
+    }
+
+    /**
+     * Set a controller class that all added embedded should use.
+     *
+     * With this, the controller for embedded routes can just be a method name.
+     * If an embedded route has a full controller (e.g. class::methodName), the
+     * controllerClass won't be applied to that route.
+     *
+     * @param string $controllerClass
+     * @return $this
+     */
+    public function setControllerClass($controllerClass)
+    {
+        if (!class_exists($controllerClass)) {
+            throw new \InvalidArgumentException(sprintf('The controller class "%s" does not exist.', $controllerClass));
+        }
+
+        $this->controllerClass = $controllerClass;
+
+        return $this;
+    }
+
+    /**
+     * Creates the final ArrayCollection, returns it, and clears everything.
+     *
+     * @return RouteCollection
+     */
+    public function flush()
```

Comment:
This name is a bit strange, most time build or get is used.
You are not really flushing anything.